integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py

```diff
@@ -59,25 +67,30 @@ class SagemakerGenerator:
     def __init__(
         self,
         model: str,
-        aws_access_key_id_var: str = "AWS_ACCESS_KEY_ID",
-        aws_secret_access_key_var: str = "AWS_SECRET_ACCESS_KEY",
-        aws_session_token_var: str = "AWS_SESSION_TOKEN",
-        aws_region_name_var: str = "AWS_REGION",
-        aws_profile_name_var: str = "AWS_PROFILE",
+        aws_access_key_id: Optional[Secret] = Secret.from_env_var(["AWS_ACCESS_KEY_ID"], strict=False),  # noqa: B008
+        aws_secret_access_key: Optional[Secret] = Secret.from_env_var(  # noqa: B008
+            ["AWS_SECRET_ACCESS_KEY"], strict=False
+        ),
+        aws_session_token: Optional[Secret] = Secret.from_env_var(["AWS_SESSION_TOKEN"], strict=False),  # noqa: B008
+        aws_region_name: Optional[Secret] = Secret.from_env_var(["AWS_DEFAULT_REGION"], strict=False),  # noqa: B008
+        aws_profile_name: Optional[Secret] = Secret.from_env_var(["AWS_PROFILE"], strict=False),  # noqa: B008
         aws_custom_attributes: Optional[Dict[str, Any]] = None,
         generation_kwargs: Optional[Dict[str, Any]] = None,
     ):
         """
         Instantiates the session with SageMaker.
 
         :param model: The name for SageMaker Model Endpoint.
-        :param aws_access_key_id_var: The name of the env var where the AWS access key ID is stored.
-        :param aws_secret_access_key_var: The name of the env var where the AWS secret access key is stored.
-        :param aws_session_token_var: The name of the env var where the AWS session token is stored.
-        :param aws_region_name_var: The name of the env var where the AWS region name is stored.
-        :param aws_profile_name_var: The name of the env var where the AWS profile name is stored.
+
+        :param aws_access_key_id: The name of the env var where the AWS access key ID is stored.
+        :param aws_secret_access_key: The name of the env var where the AWS secret access key is stored.
+        :param aws_session_token: The name of the env var where the AWS session token is stored.
+        :param aws_region_name: The name of the env var where the AWS region name is stored.
+        :param aws_profile_name: The name of the env var where the AWS profile name is stored.
```

Comment:
We should update these, they're not the name of env vars anymore.

Reply:
I will just remove it since it is omitted, or the user must pass a `Secret` - this is confusing.

I already stated in the example that the AWS credentials should be set up as environment variables or defined in a credentials file (which is then read by boto3)- should be clear enough, no?

Reply:
Maybe instead of removing it we could go with something like ```The `Secret` for AWS access key ID```?

Reply:
👍🏽